src/main/java/org/broadinstitute/consent/http/service/DarCollectionService.java

```diff
@@ -151,8 +151,9 @@ private void processDarCollectionSummariesForResearcher(List<DarCollectionSummar
       int electionCount = elections.size();
       elections.values().forEach(election -> updateStatusCount(statusCount, election.getStatus()));
       s.addAction(DarCollectionActions.REVIEW);
-      //if any DARs in the collection have approved datasets, include create progress report action
-      Set<Integer> datasetIds = dataAccessRequestDAO.findDatasetApprovalsByDars(List.copyOf(s.getReferenceIds()));
+      //if the latest DAR in the collection has at least one approved dataset,
+      //include the create progress report action
+      Set<Integer> datasetIds = dataAccessRequestDAO.findDatasetApprovalsByDar(s.getLatestReferenceId());
```

Comment:
I see that this DAO call is already in prod code, but having this in a loop will definitely impact performance. We should investigate a way to collect this information before we hit the loop.